### PR TITLE
[ORD-1873] Enable use for Ioredis, redis@3, & redi@2

### PR DESCRIPTION
**Pull request by althoff0**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,17 +21,21 @@
...
     "@types/node": "^14.14.8",
+    "@types/redis2": "npm:@types/redis@^2.8.28",
     "@typescript-eslint/eslint-plugin": "^4.8.1",
     "@typescript-eslint/parser": "^4.8.1",
     "ava": "^3.13.0",
     "eslint": "^7.13.0",
     "eslint-config-prettier": "^6.15.0",
     "eslint-plugin-prettier": "^3.1.4",
+    "ioredis": "^5.4.1",
+    "node": "^18.20.4",
```
Comment (smithamax) [inline package.json]: You don't install note via package.json, node and npm need to be installed before you can use the package.json

look in the CI workflow file

File: package.json [hunk 2]
```diff
@@ -10,8 +10,8 @@
   "scripts": {
     "test": "eslint . && ava -v",
     "lint": "eslint . ",
-    "prepare": "npm run build",
-    "build": "rm -rf ./dist && tsc"
+    "_prepare": "npm run build",
+    "_build": "rm -rf ./dist && tsc"
```
Comment (smithamax) [inline package.json]: These need to be reverted

File: index.ts [hunk 3]
```diff
@@ -114,6 +132,9 @@ export class RedisCache implements Cache {
     this.singleFlightGetCache = new Map();
     this.singleFlightApplyCache = new Map();
 
+    // Redis@3 & Redis@2 need to be promisified b/c they can't return promises (Redis@4 can)
+    // Ioredis already returns promises, so it returns the following warning when used
+    // (node:21449) [DEP0174] DeprecationWarning: Calling promisify on a function that returns a Promise is likely a mistake.
```
Comment (smithamax) [inline index.ts:137]: Awesome commenting, love it

File: ambient.d.ts [hunk 4]
```diff
@@ -0,0 +1 @@
+declare module "redis3";
```
Comment (smithamax) [inline ambient.d.ts]: Don't think we need this, we just use "redis" as redis 3 now

File: package.json (hunk 1 above)
Comment (althoff0) [reply package.json]: oh whoops thats right, forgot about that. Then should I change 12 => in the npm-publish.yml file instead (once I reverse the last commit)
```
      - uses: actions/setup-node@v1
        with:
          node-version: 12 =>18
```

File: package.json (hunk 2 above)
Comment (althoff0) [reply package.json]: thought we already did, sorry should've caught that myself

File: ambient.d.ts (hunk 4 above)
Comment (althoff0) [reply ambient.d.ts]: true

File: index.ts (hunk 3 above)
Comment (althoff0) [reply index.ts:137]: Thanks. I got excited when I upgraded to node 18 and it went away but when I did it the right way the warnings stayed. Can address this another time (maybe when we upgrade redis version or remove it and only use ioredis)

**Review by smithamax — APPROVED** [review]

**Pull request merged**